imagebuildah/stage_executor.go

```diff
@@ -1033,7 +1033,14 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			// squash the contents of the base image.  Whichever is
 			// the case, we need to commit() to create a new image.
 			logCommit(s.output, -1)
-			if imgID, ref, err = s.commit(ctx, s.getCreatedBy(nil, ""), false, s.output, s.executor.squash, lastStage); err != nil {
+			emptyLayer := false
+			if s.builder.FromImageID == "" && s.builder.FromImage == "" {
```

Comment:
We should be checking only `FromImageID` here, we don't double up on this comparison elsewhere.

Reply:
Done.